scripts/mirror_lkl_layer.py

```diff
@@ -0,0 +1,132 @@
+   limitations under the License.​
+
+    This sample mirrors LKL data from a Location Tracking Service or Location Tracking View into a standard feature service.
+    That allows the user to support dynamic joins of data.
+"""
+import argparse
+import logging
+import logging.handlers
+import traceback
+import sys
+from arcgis.gis import GIS
+from arcgis.features import FeatureLayer
+
+
+def initialize_logging(log_file=None):
+    """
+    Setup logging
+    :param log_file: (string) The file to log to
+    :return: (Logger) a logging instance
+    """
+    # initialize logging
+    formatter = logging.Formatter(
+        "[%(asctime)s] [%(filename)30s:%(lineno)4s - %(funcName)30s()][%(threadName)5s] [%(name)10.10s] [%(levelname)8s] %(message)s")
+    # Grab the root logger
+    logger = logging.getLogger()
+    # Set the root logger logging level (DEBUG, INFO, WARNING, ERROR, CRITICAL)
+    logger.setLevel(logging.DEBUG)
+    # Create a handler to print to the console
+    sh = logging.StreamHandler(sys.stdout)
+    sh.setFormatter(formatter)
+    sh.setLevel(logging.INFO)
+    # Create a handler to log to the specified file
+    if log_file:
+        rh = logging.handlers.RotatingFileHandler(log_file, mode='a', maxBytes=10485760)
+        rh.setFormatter(formatter)
+        rh.setLevel(logging.DEBUG)
+        logger.addHandler(rh)
+    # Add the handlers to the root logger
+    logger.addHandler(sh)
+    return logger
+
+
+def return_field_name(layer, name_to_check):
+    for field in layer.properties.fields:
+        if field['name'].replace("_","").lower() == name_to_check.replace("_","").lower():
+            return field['name']
+        
+        
+def main(arguments):
+    # initialize logger
+    logger = initialize_logging(arguments.log_file)
+    # Create the GIS
+    logger.info("Authenticating...")
+    # First step is to get authenticate and get a valid token
+    gis = GIS(arguments.org_url,
+              username=arguments.username,
+              password=arguments.password,
+              verify_cert=not arguments.skip_ssl_verification)
+    
+    # Get the feature layer
+    if gis.content.get(arguments.item_id):
+        logger.info("Getting feature layer")
+        item = gis.content.get(arguments.item_id)
+        mirror_layer = item.layers[0]
+        if arguments.lkl_layer_url:
+            lkl_layer = FeatureLayer(url=arguments.lkl_layer_url)
+        else:
+            logger.info("Please pass an LKL layer url!")
+            sys.exit(0)
+            
+        # Query LKL and mirror layer
+        lkl_fset = lkl_layer.query('1=1', out_sr=3857)
+        if len(lkl_fset) == 0:
+            logger.info("No LKLs in your layer yet!")
+            sys.exit(0)
+        mirror_fset = mirror_layer.query('1=1', out_sr=3857)
+        
+        add_features = []
+        update_features = []
+        logger.info("Iterating through current LKL data")
+        for feature in lkl_fset:
+            found = False
```

Comment:
Consider simplifying to:

```python
            for mirror_feature in mirror_fset:
                # use "in" instead of == comparison due to the potential for brackets to be in the GUID field
                if mirror_feature.attributes[return_field_name(mirror_layer, "global_id")].lower() in feature.attributes["globalid"].lower():
                    update_features.append(feature)
                    break
            else:
                add_features.append(feature)
```